activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -1040,6 +1056,15 @@ def binary_to_sql(limit) # :nodoc:
         end
       end
 
+      # MySQL > 5.7.5 limits lock keys to 64 chars maximum, so we use an MD5
```

Comment:
Comments about implementation details like this tend to get out of date. Let's put this in the commit message instead (if we keep it at all, see my previous comment about not needing to be this generic)